elements/lisk-p2p/src/peer_pool.ts

```diff
@@ -368,25 +334,39 @@ export class PeerPool extends EventEmitter {
 	}
 
 	public triggerNewConnections(
-		peers: ReadonlyArray<P2PDiscoveredPeerInfo>,
+		knownPeers: ReadonlyArray<P2PDiscoveredPeerInfo>,
+		fixedPeers: ReadonlyArray<P2PPeerInfo>,
 	): void {
-		const disconnectedPeers = peers.filter(
-			peer => !this._peerMap.has(constructPeerIdFromPeerInfo(peer)),
+		// Try to connect to disconnected peers avoiding the fixed ones
+		const disconnectedPeers = knownPeers.filter(
+			peer =>
+				!this._peerMap.has(constructPeerIdFromPeerInfo(peer)) ||
+				!fixedPeers.includes(peer),
 		);
 		const { outbound } = this.getPeersCountPerKind();
```

Comment:
Maybe it's out of the scope of this PR but its better if `getPeersCountPerKind()` can return `{ outboundCount, inboundCount }`, its more clear as we have other terms revolving around outbound like `outboundSocket`, `outboundShuffleInterval `, etc.